### PR TITLE
Created the Broadcasting file and constructed the different implementations

### PR DESCRIPTION
**Pull request by HeshamElsaman**
The six main functions for Broadcasting were defined and a small unit test took place.

I will add the analysis part and further modifications to the code in a separate pull request.

### CODE REVIEW

**Comment by kylemasc917** [comment]
@avgagliardo I have added you as a reviewer for this request.

**Comment by avgagliardo** [comment]
> @avgagliardo I have added you as a reviewer for this request.

Sounds good, I'll check it out asap

File: BroadcastingInMultiplication.py
Comment (avgagliardo) [inline BroadcastingInMultiplication.py:1]: Code is well-formed, executes, and comes with descriptive docstrings and code comments.

Linting returns some some PEP8 changes, none of which are critical to execution.

**Review by avgagliardo — APPROVED** [review]
Reviewed: 1 changed file

Code is well-formed, executes, and comes with descriptive docstrings and code comments.

Linting returns some some PEP8 changes, none of which are critical to execution.

**Pull request merged**